pkg/datapath/loader/loader.go

```diff
@@ -99,6 +94,15 @@ func compileAndLoad(ctx context.Context, ep endpoint, dirs *directoryInfo) error
 	return nil
 }
 
+func compileAndLoad(ctx context.Context, ep endpoint, dirs *directoryInfo) error {
```

Comment:
Almost all of the other functions are because of the endpoint. Why not make this a method of `endpoint` instead?

Reply:
I had in mind to put all BPF compilation and load logic in this package and refactor a bit to share code between the base prog setup and endpoint prog setup.